### PR TITLE
feat: generate changelog for each rule using GitHub API

### PR DESCRIPTION
**Pull request by jkodu**
Given there is a multiple repository setup, and changing how the website works to rely on the rules repo is quite some work.

I switched generating the git log to use Github API.

Closes:
- https://github.com/act-rules/act-rules-web/issues/114

Screengrabs:
image

image

### CODE REVIEW

**Comment by jkodu** [comment]
@Jym77 - I had to make so many trial commits, to understand how to pass secret token to actions. So I make a new PR again once I completed my learning process.

File: .gitignore [hunk 1]
```diff
@@ -40,7 +40,8 @@ typings/
 *.tgz
 
 # dotenv environment variables file
-.env
+.env	.env
+.env.*
```
Comment (Jym77) [inline .gitignore:44]:
```suggestion
.env*
```
Shouldn't that be enough?

File: src/templates/changelog.js [hunk 2]
```diff
@@ -40,21 +39,23 @@ const getChangelogTabulation = (changelog, url, file) => {
 		<table className="compact">
 			<thead>
 				<tr>
-					<th>Date</th>
+					<th width="100">Date</th>
+
 					<th>Description</th>
```
Comment (Jym77) [inline src/templates/changelog.js:44]: Is it voluntary that this one has no `width`?

Comment (jkodu) [reply src/templates/changelog.js:44]: Yes @Jym77 I want this column to grow in width.

**Pull request merged**